### DISCUSSION TITLE
Met maui IOS publishing

### DISCUSSION THREAD

**Discussion by waldevelop** [dotnet/maui#22974] [Q&A]
I'm using visual studio with a Net Maui project and I have a MAC connected to the same network so I can associate my PC to the MAC in order to compile the IOS application. The problem is that it takes about 2.30 hours to complete the job when I click on "publish". If I see sometimes inside my PC in the Bin/Release folder I can see the slowness to create the IPA file and surprisingly I see that the file arrives (after 20 minutes) for instance to 15 Mb and then it's cleared!!! then restart to create it to i don't know 20 Mb and then it's cleared again!! After about 1 hour and half this job is completed successfully and the the same time is necessary then to create the zip file with the same curious behaviour! My PC has an AMD 8500 processor with 32 GB of RAM and I can't imagine the reason why it's taking so much time. If I try then to distribute it to the Apple Store it takes about 3.5 hours !!! I can't work in this way. Is there any option to publish the application directly inside Xcode of MAC ? I don't find my project inside the MAC...

**Comment by davidortinau** [comment]
@waldevelop you can publish the app from Terminal on your Mac. 

https://learn.microsoft.com/en-us/dotnet/maui/ios/deployment/publish-cli?view=net-maui-8.0

Please file a Feedback issue through Visual Studio on Windows about the build issues and slowness you're experience.

**Reply by waldevelop** [reply]
Thanks for the answer, I tried what you sugested me but I received this error message

C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\17.2.8053\tools\msbuild\iOS\Xamarin.Shared.targets(2816,3): error : Cod
e signing must be enabled to create an Xcode archive. [E:\Nas\Qsync\Web\Lares\Lares.Client\Lares.Client.csproj::TargetF
ramework=net8.0-ios]

Do you know how to solve it ?
Thanks

**Comment by gabsamples6** [comment]
May be I am missing the obvious but cant you use an azure pipeline to build the ipa and send it to firebase for testing or testflight .(appcenter is dead)

**Reply by bradencohen** [reply]
Yeah, you can use Azure Pipeline or GitHub Actions. Works like a charm for us.

**Comment by alexei-lazari** [comment]
I know the problem! I have the same setup. a MAC on virtual machine (VMWARE). The problem is in SSH/SFTP speed transfer. If you try to connect to MAC from any FTP client, for ex. FileZilla - the upload to VM will be normal/super fast, but the DOWNLOAD - will be SUPER SLOW. 3-5 KB/s. It was working almost perfectly in MAC OS BIG SUR and below, but in the last 3-4 versions of MAC OS - the problem is terrible. I teste several cases and several different MAC OS versions. Clean setup from ISO. from beginning - as you try to download something from VM - the speed is extremely slow. It is the main problem for me now, I could not find a solution on internet. In my case I can wait about 20 min, and then I am using XCODE/ORGANIZER menu to distribute the app, because, if it was successful, there will be a new record. the VS will be waiting for finishing and is blocked.
